lib/v2/datasource/remote/model/balance_model.dart

```diff
@@ -3,10 +3,10 @@ import 'package:seeds/v2/domain-shared/ui_constants.dart';
 
 /// The available balance of seeds
 class BalanceModel {
-  /// Seeds available
   final double quantity;
+  final bool hasBalance;
```

Comment:
This is a hack, this is basically trying to put into the model if the network call worked or not. 
The model should not know about any of that.

Reply:
No it's not. 

It's a part of the model, the network call has _succeeded_ but there's three possibilities

1 - The user never had a balance. This is what the hasBalance flag is for. This means this user never had this token in their possession.

2 - The balance is 0 (zero). User has a balance, but it is zero. So at some point they had the token, then spent it all, and now they have 0. 

3 - The balance is > 0

That's the result of the network call. 

Could be empty = no balance, unknown user
Could be a number.

Reply:
So this was me, improving the model, since there's not just an "amount" state but there's also a has/doesn't have

Example in the user interface, we may want to show tokens where the user previously had a balance, but we definitely never want to show tokens where the user does not have a balance.

Reply:
I see, 

If the user has balance then we show the token, otherwise we dont. "0 is an acceptable balance "

Wether or not the user has this balance is not part of the balance model. The balance model knows its balance and thats it. 

If the user doesnt have this balance , then this balance model is null. 

Null represents the absence of a balance.

Reply:
I understand this is how its modeled in the chain, 

We are not bound to the chain modelings. 

But if you decide to model it like this, ill not block this pr for it.